### PR TITLE
DT-1540 Upgrade to latest gradle plugin

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/ApplicationInsightsConfiguration.kt [hunk 1]
```diff
@@ -15,16 +13,14 @@ import org.springframework.core.type.AnnotatedTypeMetadata
 @Configuration
 class ApplicationInsightsConfiguration {
   @Bean
-  @Conditional(AppInsightKeyAbsentCondition::class)
+  @ConditionalOnExpression("T(org.apache.commons.lang3.StringUtils).isBlank('\${applicationinsights.connection.string:}')")
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/ApplicationInsightsConfiguration.kt:16]: a bit ugly, but not end of the world

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/ApplicationInsightsConfiguration.kt (hunk 1 above)
Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/ApplicationInsightsConfiguration.kt:16]: I'm assuming there's a good reason Pete took this approach.

Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisontoprobation/config/ApplicationInsightsConfiguration.kt:16]: The problem is that user bean conditionals are only recommended for auto configuration classes.  There is no guarantee at the point in time that the absent condition bean will be loaded - depends on the order that spring chooses.  See https://github.com/spring-projects/spring-boot/issues/7815.  Therefore switched to conditional on expression that will always work.

**Pull request merged**